### PR TITLE
[Test] Replace rpc_wallet_tests.cpp with python RPC unit tests

### PR DESCRIPTION
**Pull request by pstratem**

### CODE REVIEW

**Comment by sipa** [comment]
Please elaborate why these tests are unnecessary.

**Comment by pstratem** [comment]
I've written python RPC tests to replace the tests.

It's much easier to maintain those tests and leads to fewer weird hacks in the setup for the tests (which is what I'm trying to avoid by removing the existing tests).

**Comment by jonasschnelli** [comment]
utACK on the additional RPC test, NACK on removing the unit test.

I don't think its wise to remove an already exiting unit test.
Also, UnitTests are on a different *level* then the RPC tests. Keep in mind: we ship the unit tests with our binaries.

**Comment by pstratem** [comment]
@jonasschnelli it's an RPC unit test though...

**Comment by jonasschnelli** [comment]
I think we need to distinct between tests on code level (linked, boost unit test framework) and test from the *outside* (RPC tests). Both are useful.

**Comment by laanwj** [comment]
ACK on removing the 'unit test'. I agree both kinds of tests are useful, however. Unit tests are supposed to test basic, low-level behavior.

E.g.:
- Wallet unit tests test the CWallet object directly.
- RPC unit tests tests basic RPC registration/dispatching.

The removed tests are clearly RPC tests disguised as unit tests. They are from the time that there wasn't a proper RPC testing framework in place yet .

**Comment by pstratem** [comment]
@laanwj I've added tests for the addmultisigaddress RPC call

File: qa/rpc-tests/wallet-accounts.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+)
+
+
+class WalletAccountsTest(BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+        self.node_args = [[], []]
```
Comment (MarcoFalke) [inline qa/rpc-tests/wallet-accounts.py]: Two nodes or one node?

File: qa/rpc-tests/wallet-accounts.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        self.nodes = start_nodes(self.num_nodes, self.options.tmpdir, self.node_args)
+        self.is_network_split = False
+
+    def run_test (self):
+        node = self.nodes[0]
+        # Check that there's no UTXO on any of the nodes
+        assert_equal(len(node.listunspent()), 0)
+        
+        node.generate(101)
+        self.sync_all()
```
Comment (MarcoFalke) [inline qa/rpc-tests/wallet-accounts.py]: You don't need sync when there is only one node.

**Comment by MarcoFalke** [comment]
> @laanwj I've added tests for the addmultisigaddress RPC call

Can you check again. I couldn't find any new addmultisig rpc tests in the added files.

**Comment by MarcoFalke** [comment]
Concept ACK 20207c0

File: qa/rpc-tests/wallet-accounts.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        
+        for account in accounts:
+            address = node.getaccountaddress(account)
+            assert(address != account_addresses[account])
+            assert_equal(node.getreceivedbyaccount(account), 2.0)
+            node.move(account, "", node.getbalance(account))
+        
+        node.generate(101)
+        
+        expected_account_balances = {"": 5200.00000000}
```
Comment (MarcoFalke) [inline qa/rpc-tests/wallet-accounts.py]: Nit: No need for float

**Comment by pstratem** [comment]
@MarcoFalke nits picked

**Comment by jonasschnelli** [comment]
Agree with @laanwj.
utACK https://github.com/bitcoin/bitcoin/pull/8450/commits/9578333ec456bbfaccfecf9ea561c421f3a5d547

**Pull request merged**